resources-generator/src/main/kotlin/dev/icerock/gradle/data/ExtractingBaseLibraryImpl.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2025 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.gradle.data
+
+import org.jetbrains.kotlin.konan.file.File
+import org.jetbrains.kotlin.konan.file.file
+import org.jetbrains.kotlin.konan.file.unzipTo
+import org.jetbrains.kotlin.konan.file.withZipFileSystem
+import org.jetbrains.kotlin.library.KotlinLibraryLayout
+import org.jetbrains.kotlin.library.impl.KotlinLibraryLayoutImpl
+
+open class ExtractingKotlinLibraryLayout(zipped: KotlinLibraryLayoutImpl) : KotlinLibraryLayout {
```

Comment:
better to mark it internal. and also add link to original source code (to kotlin repo as permalink)